### PR TITLE
sign in and test

### PR DESCRIPTION
**Pull request by DiegoWolox**
## Summary

Sign in endpoing and unit test

## Trello Card

[sign in](https://trello.com/c/RLrGdxQh/2-sign-in)

### CODE REVIEW

File: app/middlewares/users.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const findUser = async req => {
+  logger.info(`trying get user: ${req.body.email}`);
+  const userFound = await userService.findByEmail(req.body.email);
+  if (!userFound) throw errors.notFound('invalid email or password');
+  return userFound;
+};
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const userFound = await findUser(req);
```
Comment (marioportela1993) [inline app/middlewares/users.js]: Don't pass the entire req value, pass only the email instead

File: app/middlewares/users.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    await checkPassword(userFound, req);
```
Comment (marioportela1993) [inline app/middlewares/users.js]: Same comment, pass only the password. Maybe could be a good idea destructuring the email and the password previously.

File: app/models/user.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
-const { hash } = require('../helpers/crypto');
+const { hash, compare } = require('../helpers/crypto');
 
 const hashPassword = async user => {
```
Comment (marioportela1993) [inline app/models/user.js:3]: maybe this function should be a helper. What do you think?

File: app/models/user.js [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
   const { password } = user;
   const hashedPassword = await hash(password);
   Object.assign(user, { password: hashedPassword });
 };
 
+function isCorrectPassword(enteredPassword) {
```
Comment (marioportela1993) [inline app/models/user.js:9]: Same comment

File: config/index.js [hunk 5]
```diff
@@ -44,6 +44,9 @@ const config = {
       apiDate: process.env.API_DATE || 'X-API-Date',
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    },
+    token: {
```
Comment (marioportela1993) [inline config/index.js]: You can use the `session.secret` value created in the line 39 instead

**Review by marioportela1993 — CHANGES_REQUESTED** [review]
Excellent work! check the comments and apply them

File: app/middlewares/users.js (hunk 1 above)
Comment (DiegoWolox) [reply app/middlewares/users.js]: changed

File: app/middlewares/users.js (hunk 2 above)
Comment (DiegoWolox) [reply app/middlewares/users.js]: changed

File: app/models/user.js (hunk 3 above)
Comment (DiegoWolox) [reply app/models/user.js:3]: that function only affects the users model. A utility for hashing strings already exists in helpers file

File: app/models/user.js (hunk 4 above)
Comment (DiegoWolox) [reply app/models/user.js:9]: in this case I need a function declared in the old way (function) the arrow notation does not work for me

File: config/index.js (hunk 5 above)
Comment (DiegoWolox) [reply config/index.js]: changed

File: app/models/user.js (hunk 4 above)
Comment (marioportela1993) [reply app/models/user.js:9]: Ok let the function in the model

**Review by marioportela1993 — APPROVED** [review]

**Pull request merged**